modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/distruptor/inbound/DisruptorBasedInboundEventManager.java

```diff
@@ -170,11 +170,29 @@ public void clearMessagesFromQueueInMemory(String storageQueueName, Long purgedT
     }
 
     @Override
-    public void purgeQueue(String destinationQueue, String ownerName, boolean isTopic) throws AndesException {
+    public int purgeQueue(String destinationQueue, String ownerName, boolean isTopic) throws AndesException {
+
+        QueuePurgeEvent purgeEvent = new QueuePurgeEvent(destinationQueue, ownerName, isTopic);
+        publishToRingBuffer(InboundEvent.Type.QUEUE_PURGE_EVENT, purgeEvent,
+                "Queue purge event with destination: " + destinationQueue + " owner: " + ownerName + " isTopic: " +
+                        isTopic + " published to Disruptor");
+
+        try {
+            return purgeEvent.getPurgedCount();
+        } catch (ExecutionException e) {
+            throw new AndesException("Exception occurred while purging destination " + destinationQueue, e);
+        } catch (InterruptedException e) {
+            // restore interrupted status for this thread.
+            Thread.currentThread().interrupt();
+        }
+        return -1; // error
```

Comment:
Should we be defining -1 as an Constant? (better may be a enum? ) (- I'm open to discussion)